engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -93,550 +104,349 @@
-     * @param z the z coordinate to set the size; must be > 0
-     * @return this after modification
-     */
-    public BlockRegion setSize(int x, int y, int z) {
-        Preconditions.checkArgument(x > 0);
-        Preconditions.checkArgument(y > 0);
-        Preconditions.checkArgument(z > 0);
-        this.maxX = this.minX + x - 1;
-        this.maxY = this.minY + y - 1;
-        this.maxZ = this.minZ + z - 1;
-        return this;
+    public BlockRegion maxZ(int z) {
+        return this.maxZ(z, this);
     }
 
-    /**
-     * Set the size of the block region from minimum the minimum corner.
-     *
-     * @param size the size to set; all dimensions must be > 0
-     * @return this after modification
-     */
-    public BlockRegion setSize(Vector3ic size) {
-        return setSize(size.x(), size.y(), size.z());
+    @Override
+    public BlockRegion setMax(int x, int y, int z, BlockRegion dest) {
+        return dest.set(this.minX, this.minY, this.minZ, maxX, maxY, maxZ);
     }
 
-    /**
-     * the number of blocks for the +x, +y, +z from the minimum to the maximum
-     *
-     * @param dest will hold the result
-     * @return dest
-     */
-    public Vector3i getSize(Vector3i dest) {
-        return dest.set(sizeX(), sizeY(), sizeZ());
+    public BlockRegion setMax(int x, int y, int z) {
+        return this.setMax(x, y, z, this);
     }
 
-    /**
-     * The number of blocks on the X axis
-     *
-     * @return number of blocks in the X axis
-     */
-    public int sizeX() {
-        return this.maxX - this.minX + 1;
+    public BlockRegion setMax(Vector3ic max) {
+        return this.setMax(max, this);
     }
 
-    /**
-     * The number of blocks on the Y axis
-     *
-     * @return number of blocks in the Y axis
-     */
-    public int sizeY() {
-        return this.maxY - this.minY + 1;
+    public BlockRegion addToMax(int dx, int dy, int dz) {
+        return this.addToMax(dx, dy, dz, this);
     }
 
-    /**
-     * The number of blocks on the Z axis
-     *
-     * @return number of blocks in the Z axis
-     */
-    public int sizeZ() {
-        return this.maxZ - this.minZ + 1;
+    public BlockRegion addToMax(Vector3ic dmax) {
+        return this.addToMax(dmax, this);
     }
 
-    // -- world ------------------------------------------------------------------------------------------------------//
-
-    //TODO: 1.9.26 has a constant interface for aabbf
-    public AABBf getBounds(AABBf dest) {
-        dest.minX = minX - .5f;
-        dest.minY = minY - .5f;
-        dest.minZ = minZ - .5f;
+    // -- size -------------------------------------------------------------------------------------------------------//
 
-        dest.maxX = maxX + .5f;
-        dest.maxY = maxY + .5f;
-        dest.maxZ = maxZ + .5f;
+    @Override
+    public BlockRegion setSize(int x, int y, int z, BlockRegion dest) {
+        return dest.set(this.minX, this.minY, this.minZ, this.minX + x, this.minY + y, this.minZ + z);
+    }
```

Comment:
I would drop this as well. just as easy to just do `new BlockRegion(a).setSize(3,3,3)` or `b.set(a).setSize(3,3,3)` vs `a.setSize(3,3,3, new BlockRegion(BlockRegion.INVALID));`

Reply:
> A setter that sets the value of another object is kind of strange?

I'm REALLY confused by now o.O At first I was confused by this pattern in JOML, than you and others explained to me why that's a good thing. Now, it seems the other way around, and we should probably not offer those `dest` variants at all...

In all cases, these two expressions are equivalent:
```java
a.op(b, dest) == dest.set(a).op(b)
```

Reply:
Drop the dest variations for the setters. They're simple enough for the most part not to need it. I guess that can be debatable.